code/drasil-example/dblpendulum/lib/Drasil/DblPendulum/DataDefs.hs

```diff
@@ -34,7 +34,7 @@ positionXDD_1 = ddENoRefs positionXQD_1 Nothing "positionXDD1" [positionXRef_1,
 positionXQD_1 :: SimpleQDef
 positionXQD_1 = mkQuantDef xPos_1 positionXEqn_1
 
-positionXEqn_1 :: Expr
+positionXEqn_1 :: ExprC r => r
```

Comment:
I must admit that I'm not at all sure about having this much polymorphism in the examples themselves.

In some sense, it's ok, as we should be moving a lot of this to an external DSL, and that will hide all of this. Not sure how soon that'll happen, so in the meantime, the examples will look more complex.

Reply:
Would we be able to create a type synonym that hides the polymorphism a bit? If so, would that change anything with this issue?

Reply:
Not easily? Part of the point is that this is polymorphic, so that we do need/want a variable showing up.

Maybe a GADT, like
```haskell
data PExpr r where
  E :: ExprC r => r -> PExpr r
```
would let you replace `ExprC r => r` with `PExpr r` (which could be renamed `Expr r` if we want).  But we'd then have to have an easy way to add that `E` constructor.  Which is a lot like `express`!

Kind of feels like moving the deck chairs on the Titanic... but less bad.

Reply:
I see what you mean. What about with `RankNTypes`?
```
type PExpr = forall r . ExprC r => r
```

Reply:
That might work!  I'd forgotten about that.

Reply:
I just updated this, it looks like it worked. What do you think?

Reply:
I think we've converged. Excellent.